platform/viewer/src/connectedComponents/ConnectedUserPreferencesModal.js

```diff
@@ -0,0 +1,59 @@
+import { connect } from 'react-redux';
+import { UserPreferencesModal } from '@ohif/ui';
+import OHIF from '@ohif/core';
+import { hotkeysManager } from './../App.js';
+import cloneDeep from 'lodash.clonedeep';
+
+const { setUserPreferences } = OHIF.redux.actions;
+
+const mapStateToProps = (state, ownProps) => {
+  const isEmpty = obj => Object.keys(obj).length === 0;
+  const newHotKeysData =
+    state.preferences && !isEmpty(state.preferences.hotKeysData)
+      ? state.preferences.hotKeysData
+      : hotkeysManager.hotkeyDefinitions;
+  hotkeysManager.setHotkeys(hotkeysManagerFormatter(cloneDeep(newHotKeysData)));
+  return {
+    isOpen: ownProps.isOpen,
+    windowLevelData: state.preferences ? state.preferences.windowLevelData : {},
+    hotKeysData: newHotKeysData,
+  };
+};
+
+/* HotkeysManager's hotkeysDefinitions have different output/input. */
+const hotkeysManagerFormatter = hotKeysData => {
```

Comment:
Can we move this method to hotkeyManager file instead?

Reply:
Done! thx.